src/Service/VendorService/OverDrive/Api/Client.php

```diff
@@ -0,0 +1,316 @@
+     * @throws InvalidArgumentException
+     */
+    public function getProducts(int $limit, int $offset): array
+    {
+        $response = $this->httpClient->request('GET', $this->getProductsEndpoint(), [
+            'headers' => $this->getHeaders(),
+            'query' => ['limit' => $limit, 'offset' => $offset],
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->products;
+    }
+
+    /**
+     * Get the total number of products from the overdrive api ('totalItems' field in the response).
+     *
+     * @return int
+     *   The total number of products
+     *
+     * @throws AuthException
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     */
+    public function getTotalProducts(): int
+    {
+        $response = $this->httpClient->request('GET', $this->getProductsEndpoint(), [
+            'headers' => $this->getHeaders(),
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->totalItems;
+    }
+
+    /**
+     * Get the headers for OverDrive api calls.
+     *
+     * @return string[]
+     *   Array of headers
+     *
+     * @throws AuthException
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     */
+    private function getHeaders(): array
+    {
+        return [
+            'User-Agent' => self::USER_AGENT,
+            'Content-Type' => 'application/json',
+            'Authorization' => $this->getAuthorization(),
+        ];
+    }
+
+    /**
+     * Get the products endpoint for the account used.
+     *
+     * @return string
+     *   The complete URI for the products endpoint
+     *
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     * @throws AuthException
+     */
+    private function getProductsEndpoint(): string
+    {
+        if (empty($this->productsEndpoint)) {
+            $this->productsEndpoint = $this->fetchProductsEndpoint();
+        }
+
+        return $this->productsEndpoint;
+    }
+
+    /**
+     * Fetch the products endpoint for the account used from the Library API.
+     *
+     * @see https://developer.overdrive.com/apis/library-account
+     *
+     * @return string
+     *   The complete URI for the products endpoint
+     *
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     * @throws AuthException
+     */
+    private function fetchProductsEndpoint(): string
+    {
+        $item = $this->cache->getItem('overdrive.api.productsEndpoint');
+
+        if ($item->isHit()) {
+            $endpoint = $item->get();
+        } else {
+            $response = $this->httpClient->request('GET', $this->libraryAccountEndpoint, [
+                    'headers' => $this->getHeaders(),
+            ]);
+
+            $content = $response->getBody()->getContents();
+            $json = json_decode($content, false);
```

Comment:
Same here regrading the json and `false` and missing check that the body exists (status code).